pkg/controller/scylladbdatacenter/resource.go

```diff
@@ -2116,3 +2087,21 @@ func MakeUpgradeContextConfigMap(sdc *scyllav1alpha1.ScyllaDBDatacenter, uc *int
 		},
 	}, nil
 }
+
+func makePropagableLabels(sdc *scyllav1alpha1.ScyllaDBDatacenter) map[string]string {
+	labels := map[string]string{}
+	maps.Copy(labels, sdc.Labels)
+	for _, k := range nonPropagatedLabelKeys {
+		delete(labels, k)
+	}
+	return labels
+}
+
+func makePropagableAnnotations(sdc *scyllav1alpha1.ScyllaDBDatacenter) map[string]string {
+	annotations := map[string]string{}
+	maps.Copy(annotations, sdc.Annotations)
+	for _, k := range nonPropagatedAnnotationKeys {
+		delete(annotations, k)
+	}
+	return annotations
+}
```

Comment:
Duplicate logic (DRY) - suggestion: refactor into a generic function and inject the values as appropriate:

```go
mapCopyExcludingKeys(sdc.Annotations, nonPropagatedAnnotationKeys)
mapCopyExcludingKeys(sdc.Labels, nonPropagatedLabelKeys)
```

Reply:
done

Reply:
> I'll amend the code to just inline the stdlib functions then.

This misses the point of my suggestion, which is:

- maintaining the fact that "creating a copy with elements filtered out" is a single function call
- not duplicating logic if not necessary.

I would judge that keeping a function as generic as necessary for the reasonable set of call sites is enough - unless we're fixing this across a wide range of applications/places, I think that operating on `map[string]string` and `[]string` should be just fine.

Therefore my suggestion is to either:
- apply the original suggestion
- keep the original as it was

I am under impression that inlining the stdlib functions satisfies no one (you included).